sdk/core/azure-core/CMakeLists.txt

```diff
@@ -26,12 +26,9 @@ include(CreateMapFile)
 find_package(Threads REQUIRED)
 
 if(BUILD_TRANSPORT_CURL)
-  # min version for `CURLSSLOPT_NO_REVOKE`
-  # https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
-  set(CURL_MIN_REQUIRED_VERSION 7.44)
-  find_package(CURL ${CURL_MIN_REQUIRED_VERSION} CONFIG QUIET)
+  find_package(CURL CONFIG QUIET)
```

Comment:
We should reinvestigate our curl min requirement (and probably if-def in this case the one feature that is forcing us to move to 7.77).

Otherwise, we are moving to a version that isn't available out-of-the-box on Ubuntu 20.04.

> cURL 7.58.0 is the most recent and secure version for Ubuntu 18.04.
cURL 7.68.0 is the most recent and secure version for Ubuntu 20.04.
cURL 7.81.0 is the most recent and secure version for Ubuntu 22.04.

https://launchpad.net/ubuntu/focal/+source/curl

Reply:
I removed the dependency version upgrade from this PR, just moved the check to different place.

Reply:
FYI, the curl version on WSL for Ubuntu 20.04.6 LTS is also 7.68.0.
> Ubuntu 20.04.6 LTS (GNU/Linux 5.15.90.1-microsoft-standard-WSL2 x86_64)
```text
curl --version
curl 7.68.0 (x86_64-pc-linux-gnu) libcurl/7.68.0 OpenSSL/1.1.1f zlib/1.2.11 brotli/1.0.7 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh/0.9.3/openssl/zlib nghttp2/1.40.0 librtmp/2.3
Release-Date: 2020-01-08
```